src/Core/Datatypes/Geometry.h

```diff
@@ -57,14 +57,16 @@ namespace Datatypes
     //weird signature--need to avoid raw ptrs and shared_from_this for now
     virtual void addToList(GeometryBaseHandle handle, GeomList& list);
 
-    const std::string& uniqueID() const { return objectName_; }
+    const std::string uniqueID() const { return objectName_ + idIncrement_; }
+    void changeID() { idIncrement_ = std::to_string(std::stoi(idIncrement_)+1); }
```

Comment:
@RubioJr9 @Haydelj YANRQ (yet another naive renderer question): for this incrementing of object id, do you think we should use an atomic object? Meaning, could multiple ViewScenes containing the same object introduce a race for incrementing the id? In that case, we'd have undefined ids, which would be bad.

Reply:
For instance, what if the same widget were in two ViewScenes, and they were linked, and the user clicked one while auto-spinning another, or something like that...

Reply:
I think there could be a race condition if a mouse press and release is being done in a linked viewscene to get the widget to move in both viewers, but I doubt Jacob set it up that way because that would fail if the camera wasn't the exact same and handling window sizes would be messy. Regardless, it wouldn't hurt to use an atomic just in case.

Reply:
Yes, the viewSceneLocking code shouldn't interact with the widget selection code at all